py/src/braintrust/logger.py

```diff
@@ -459,13 +472,20 @@ def __init__(self, log_conn: LazyValue[HTTPConnection]):
 
     def log(self, *args):
         self._start()
+        did_notify_drop = False
         for event in args:
             try:
                 self.queue.put_nowait(event)
             except queue.Full:
                 # Notify consumers to start draining the queue.
                 self.queue_filled_semaphore.release()
-                self.queue.put(event)
+                if not self.queue_drop_when_full:
+                    self.queue.put(event)
+                elif not did_notify_drop:
+                    _logger.warning(
+                        f"Queue is currently full at size {self.queue_maxsize}. Dropping additional elements"
```

Comment:
can we print ho w many were dropped, and same comment, if dropped can we save them to the filesystem?